### DISCUSSION TITLE
found page without a React Component as default export

### DISCUSSION THREAD

**Discussion by JwngmaBasumatary** [vercel/next.js#45163] [Help]
This happening on a dynamic route in my pages folder. The program runs locally without any problem but this error comes when I deploy it on vercel. I have tried every different possible combination of exports but this happens every time.

My File : pages/products/[productId].js

```
import Head from "next/head";
import React from "react";
import ProductDetails from "../../components/Product/ProductDetails/ProductDetails";

export default function Index() {
  return <ProductDetails />;
}
```

Why is this happening? I have tried all the answers on Stack Overflow and even posted my own question but I have no solutions yet.

Edit : I am sharing my ProductDetails.jsx component and the tree view  of my project as requested.

This is an image of my ProductDetails Page : 

![image](https://user-images.githubusercontent.com/97206612/214001599-17bfabb3-1317-406e-a015-aa7c9378f2c7.png)

And this is the root folder structure : 

![image](https://user-images.githubusercontent.com/97206612/214001261-70797dd9-2605-4c5e-ae78-fd33849c6a90.png)

**Comment by icyJoseph** [comment]
Got a repository to look at?

Could you share how's ProductDetails defined and exported?

Where does `components` live? Could you share a tree view of your files?

**Reply by JwngmaBasumatary** [reply]
I have updated my question. The files are too big so I have uploaded an image to give you an idea of the structure. Let me know if you need any more information. I can't share the repository as this is a private project for a client.

**Reply by icyJoseph** [reply]
Alright last questions, I hope 😅 , are you sure all other pages have a default export? Why did you arrive to the conclusion that the error is within this one? 

And, are you using something like `next export`?

**Reply by JwngmaBasumatary** [reply]
It happens while I am deploying on Vercel. Locally, everything works fine but while deploying on vercel it shows this error in this very page.

**Reply by icyJoseph** [reply]
That's very strange to say the least, do you see the same issue if you run `npm run build` locally? 

The page file is `.jsx` correct? And there's no additional files inside pages, everything that's inside pages must have a default export.
